src/actions.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Jenkins charm actions."""
+
+import ops
+
+import jenkins
+from state import State
+
+
+class Observer(ops.Object):
```

Comment:
That looks a little bit different from what is recommended by the Charm Complexity guide.
Flask, for example, observe action event in the charm logic
https://github.com/canonical/flask-k8s-operator/blob/15056fd2451a5279ab45880dee5fa6fca315bae0/src/charm.py#L94

Reply:
I'm not sure I can find actions under [Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619) document, perhaps have I missed something?

As of my reasoning, I tried to make things as modular as possible so that it's smaller in scope. One of the problems I'm facing now is a huge `jenkins.py` file which i had to slap on a `# pylint: ignore`. Would that be a good enough reasoning to make this into a separate module? I dont' mind moving actions back into the main `charm.py` :)

Thank you for the review, +1 review :coin:

Reply:
I would go for leaving in the charm. What do you think @jdkandersson ?

Reply:
It seems reasonable to have an actions observer to reduce the amount of code in `charm.py`. We don't need to do it everywhere, e.g., if `charm.py` is quite short (like 100-200 lines), the action observation can be done there. If it is starting to get complex, this can be one of the ways that code can be moved to a separate module

Reply:
Considering the last answer, I understood that in this case this should be moved to charm.py, right?

Reply:
I'll leave that to you and Yanks to decide 🙂

Reply:
No worries then :)